src/features/core/components/funnel-step.tsx

```diff
@@ -0,0 +1,96 @@
+import { PropsWithChildren, useRef, useState } from 'react';
+
+import { cn } from '../utils/cn';
+
+import { BackButton } from './back-button';
+import { LoadingOverlay } from './loading-overlay';
+
+interface FunnelStepProps {
+  title?: string;
+  stepTitle: string;
+  description?: string;
+  button?: React.ReactNode;
+  hideUndo?: boolean;
+  loading?: boolean;
+}
+
+// TODO: 버튼 width를 w-full에서 w-fit로 바꾸면서 이상해진거 고치기
+
+export function FunnelStep({
+  title,
+  stepTitle,
+  description,
+  button,
+  hideUndo = false,
+  loading = false,
+  children,
+}: PropsWithChildren<FunnelStepProps>) {
+  const [scrollAmount, setScrollAmount] = useState(0);
+  const scrollRef = useRef<HTMLDivElement>(null);
+
+  const handleScroll = () => {
+    if (scrollRef.current) setScrollAmount(scrollRef.current.scrollTop);
+  };
```

Comment:
p4; 이게 첫 상태에서도 호출 되는 함수는 아니다 보니까 좀 애매한 것 같아요.

```suggestion
  const scrollRef = useCallback((div: HTMLDivElement) => {
    const handler => setScrollAmount(handler.scrollTop);
    div.addEventListener('scroll', handler);
    return () => {
      div.removeEventListener('scroll', handler);
    }
  }, []);
```
하고 `<div ref={scrollRef}>`